### PR TITLE
Add holdout evaluator

### PR DESCRIPTION
**Pull request by bfolie**
# Citrine Python PR

## Description 
Please briefly explain the goal of the changes/this PR.
The reviewer should be able to understand why the change is being made by reading this description
and its links (e.g. JIRA tickets).

### PR Type:
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Maintenance (non-breaking change to assist developers)

### Adherence to team decisions
- [x] I have added tests for 100% coverage
- [x] I have written Numpy-style docstrings for every method and class.
- [ ] I have communicated the downstream consequences of the PR to others.
- [x] I have bumped the version in setup.py

### CODE REVIEW

**Comment by bfolie** [comment]
This is done, but can't be merged until the relevant [backend PR](https://github.com/CitrineInformatics/platform-backend/pull/2647/files) is finished

File: src/citrine/informatics/predictor_evaluation_result.py [hunk 1]
```diff
@@ -220,3 +221,41 @@ def responses(self) -> Set[str]:
...
+
+    Results held-out response can be accessed via ``result['response_name']``,
+    where ``result`` is a
+    :class:`citrine.informatics.predictor_evaluation_result.HoldoutSetResult`
+    and ``'response_name'`` is a response analyzed by a
+    :class:`citrine.informatics.predictor_evaluator.HoldoutSetEvaluator`.
+
+    """
+
+    _evaluator = properties.Object(PredictorEvaluator, "evaluator")
```
Comment (sfriedowitz) [inline src/citrine/informatics/predictor_evaluation_result.py:237]: Should the evaluator be specific as a `HoldoutSetEvaluator` here ? Same goes for the CV result above.

I think the backend has these typed specifically. but I'm unsure if its straightforward to check that upon deserialization here.

**Review by sfriedowitz — COMMENTED** [review]
One quick question, but looking good.

**Comment by bfolie** [comment]
Blocked on https://citrine.atlassian.net/browse/PLA-9920

File: src/citrine/informatics/predictor_evaluation_result.py (hunk 1 above)
Comment (bfolie) [reply src/citrine/informatics/predictor_evaluation_result.py]: I made it more specific. But also, the `type` field indicates whether it's a `HoldoutSetEvaluator` or a `CrossValidationEvaluator`.

**Review by genfx999 — APPROVED** [review]

**Pull request merged**